### PR TITLE
add support function to bootstrap the ledgers of a case

### PR DESCRIPTION
**Pull request by czue**
this is a short term workaround for a mobile bug that is getting fixed in 2.23

### CODE REVIEW

File: corehq/apps/reports/views.py [hunk 1]
```diff
@@ -1040,6 +1041,25 @@ def resave_case(request, domain, case_id):
...
 @require_POST
+def bootstrap_ledgers(request, domain, case_id):
+    case = get_document_or_404(CommCareCase, domain, case_id)
+    if (not StockTransaction.objects.filter(case_id=case_id).exists() and
+            SQLProduct.objects.filter(domain=domain).exists()):
+        submit_case_blocks([
+            '''<balance xmlns="http://commcarehq.org/ledger/v1" entity-id="{case_id}" date="{date}" section-id="stock">
+           <entry id="{product_id}" quantity="0" />
+        </balance>'''.format(
+            date=json_format_datetime(datetime.utcnow()),
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/reports/views.py [hunk 2]
```diff
@@ -1040,6 +1041,25 @@ def resave_case(request, domain, case_id):
...
+            case_id=case_id,
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/reports/views.py [hunk 3]
```diff
@@ -1040,6 +1041,25 @@ def resave_case(request, domain, case_id):
...
+            product_id=SQLProduct.objects.filter(domain=domain).values_list('product_id', flat=True)[0]
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/reports/views.py [hunk 4]
```diff
@@ -1040,6 +1041,25 @@ def resave_case(request, domain, case_id):
...
+        )], domain=domain)
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/reports/views.py (hunk 1 above)
Comment (czue) [inline corehq/apps/reports/views.py]: not sure how better to format this. open to suggestions, but otherwise going to ignore.

**Comment by czue** [comment]
(up for grabs, :white_check_mark: )

**Comment by benrudolph** [comment]
:+1:

**Comment by snopoke** [comment]
@czue I assume you'll remove this after that bug gets fixed? might be worth just adding a comment to the function to indicate that as well in case you forget.

**Comment by czue** [comment]
good call https://github.com/dimagi/commcare-hq/pull/7949

**Pull request merged**